File: 13.RomantoInteger.js
var romanToInt = function (s) {
  const romans = new Map();
  romans.set('I', 1);
  romans.set('V', 5);
  romans.set('X', 10);
  romans.set('L', 50);
  romans.set('C', 100);
  romans.set('D', 500);
  romans.set('M', 1000);

  let result = romans.get(s[s.length - 1]);
  for (let i = s.length - 2; i >= 0; i--) {
    if (romans.get(s[i]) >= romans.get(s[i + 1])) {
      result = result + romans.get(s[i]);
    } else {
      result = result - romans.get(s[i]);
    }
  }
  return result;
};

File: 13.RomantoInteger.java
class Solution {
  public int romanToInt(String s) {
    Map<Character, Integer> romans = new HashMap<>();
    romans.put('I', 1);
    romans.put('V', 5);
    romans.put('X', 10);
    romans.put('L', 50);
    romans.put('C', 100);
    romans.put('D', 500);
    romans.put('M', 1000);

    int result = romans.get(s.charAt(0));
    int prev = 0;

    for (int i = 1; i < s.length(); i++) {
      if (romans.get(s.charAt(prev)) < romans.get(s.charAt(i))) {
        result = result - romans.get(s.charAt(prev)) * 2;
      }
      result = result + romans.get(s.charAt(i));
      prev = i;
    }

    return result;
  }
}